src/main/java/net/neoforged/neoforge/common/IOUtilities.java

```diff
@@ -0,0 +1,160 @@
+/*
+ * Copyright (c) NeoForged and contributors
+ * SPDX-License-Identifier: LGPL-2.1-only
+ */
+
+package net.neoforged.neoforge.common;
+
+import java.io.BufferedOutputStream;
+import java.io.DataOutputStream;
+import java.io.IOException;
+import java.io.OutputStream;
+import java.nio.file.Files;
+import java.nio.file.OpenOption;
+import java.nio.file.Path;
+import java.nio.file.StandardCopyOption;
+import java.nio.file.StandardOpenOption;
+import java.nio.file.attribute.BasicFileAttributes;
+import java.util.function.BiPredicate;
+import java.util.function.Consumer;
+import net.minecraft.nbt.CompoundTag;
+import net.minecraft.nbt.NbtIo;
+import org.jetbrains.annotations.Nullable;
+
+/**
+ * Declares a class containing helpers for performing file I/O in a resillient
+ * manner.
+ */
+public final class IOUtilities {
```

Comment:
Should this be marked `@ApiStatus.Internal`?

Should we also rename this to something like `ResilientIOUtilities`, to distinguish it from normal IO utilities and keeping in line with the javadocs?

Reply:
My thought was that it would be a convenient utility for other modders too, hence why it was made public.

Reply:
I think it's fine to have public.